pkgs/applications/science/math/gap/default.nix

```diff
@@ -1,36 +1,32 @@
-{ stdenv, fetchurl, pari ? null }:
+{ stdenv, fetchurl, m4, gmp }:
 
 let
   baseName = "gap";
-  version = "4r4p12";
+  version = "4r8p3";
 
-  pkgVer = "2012_01_12-10_47_UTC";
-  pkgSrc = fetchurl {
-    url = "ftp://ftp.gap-system.org/pub/gap/gap4/tar.bz2/packages-${pkgVer}.tar.bz2";
-    sha256 = "0z9ncy1m5gvv4llkclxd1vpcgpb0b81a2pfmnhzvw8x708frhmnb";
-  };
+  pkgVer = "2016_03_19-22_17";
 in
 
 stdenv.mkDerivation rec {
   name = "${baseName}-${version}";
 
   src = fetchurl {
-    url = "ftp://ftp.gap-system.org/pub/gap/gap4/tar.gz/${baseName}${version}.tar.gz";
-    sha256 = "0flap5lbkvpms3zznq1zwxyxyj0ax3fk7m24f3bvhvr37vyxnf40";
+    url = "ftp://ftp.gap-system.org/pub/gap/gap48/tar.gz/${baseName}${version}_${pkgVer}.tar.gz";
+    sha256 = "1rmb0lj43avv456sjwb7ia3y0wwk5shlqylpkdwnnqpjnvjbnzv6";
   };
 
-  buildInputs = [ pari ];
-
+  configureFlags = [ "--with-gmp=system" ];
+  buildInputs = [ m4 gmp ];
+  
+  postBuild = ''
+    (cd pkg && chmod +x ../bin/BuildPackages.sh && ../bin/BuildPackages.sh)
```

Comment:
I think it would be better to break this out as:

```
chmod +x bin/BuildPackages.sh
pushd pkg
../bin/BuildPackages.sh
popd
```

instead of one-lining it.I'm concerned about the `()` swallowing errors.